rpc/internal/service.go

```diff
@@ -241,17 +241,17 @@ func (srv *perfService) addArtifacts(record *model.PerformanceResult, artifacts
 }
 
 func (srv *perfService) addFTDCRollupsJob(id string, artifacts []model.ArtifactInfo) error {
-	ftdc := false
+	var hasEventData bool
 
 	for _, artifact := range artifacts {
 		if artifact.Schema != model.SchemaRawEvents {
 			continue
 		}
 
-		if ftdc {
+		if hasEventData {
 			return errors.New("cannot have more than one raw events artifact")
 		}
-		ftdc = true
+		hasEventData = true
 
 		job, err := units.NewFTDCRollupsJob(id, &artifact)
```

Comment:
can we not call getRemoteQueue on every loop and put that outside?

Reply:
we can, but either way it will only get called once since we only ever create one job (it errors if it finds more than one ftdc artifact)

Reply:
moved it